website/docs/core/architecture.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Architecture
+---
+
+authentik consists of a handful of compoennts, most of which are required for a functioning setup.
+
+```mermaid
+graph LR
+    user(User) --> ak_server(authentik Server)
+    ak_server --> ak_server_core(authentik Server Core)
+    ak_server --> ak_outpost(Embedded outopst)
+    ak_server_core --> db(PostgreSQL)
+    ak_server_core --> cache(Redis)
+    ak_worker(Background Worker) --> db(PostgreSQL)
+    ak_worker(Background Worker) --> cache(Redis)
+```
+
+### Server
+
+The server container consists of two sub-components, the actual server itself and the embedded outpost. Incoming requests to the server container(s) are routed by a lightweight router to either the _Core_ server or the embedded outpost. This router also handles requests for any static assets such as JavaScript and CSS files.
+
+##### Core
+
+The core sub-component handles most of authentik's logic, such as API requests, flow executions, any kind of SSO requests, etc.
+
+##### Embedded outpost
+
+Similar to [other Outposts](../outposts/index.mdx), this outposts allows using [Proxy providers](../providers/proxy/index.md) without deploying a separate outpost
+
+##### Persistence
+
+-   `/media` is used to store icons and such, but not required, and if not mounted, authentik will allow you to set a URL to icons in place of a file upload
```

Comment:
```suggestion
-   The `/media`directory is used to store icons and such, but not required, and if not mounted, authentik will allow you to set a URL to icons instead of a file upload.
```

Reply:
By "set a URL to icons" do you mean "define a URL for a clickable icon" or "define a URL to display an icon instead of uploading a file"?

Reply:
The title of this section is Persistence, but the sentence in it is about a /media directory.... how are they related, how does the directory help with persistence (and of what?)?

Maybe turn this section into a Note? All of the other headers reflect a component shown in the diagram, but not this one about persistence.

Reply:
Oh... I see there is a Persistence section for several of the components. Yes, I would turn these into Notes.

If you want to keep them as headings, I;d suggest for SEO reasons to retitle them to "Persistence in an outpost" and "Persistence in a background worker", etc. So that we don't end up with a bunch of heading s that all say same thing.